mediorum/server/delist_statuses.go

```diff
@@ -0,0 +1,275 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"golang.org/x/exp/slog"
+
+	"mediorum/ethcontracts"
+	"mediorum/server/signature"
+)
+
+// Constants
+const DelistStatusPollingInterval = 20 * time.Second
+const HTTPTimeout = 5 * time.Minute
+const DelistBatchSize = 5000
+
+type (
+	TrackDelistStatus struct {
+		CreatedAt time.Time `json:"-"`
+		TrackID   int       `json:"trackId"`
+		OwnerID   int       `json:"ownerId"`
+		TrackCID  string    `json:"trackCid"`
+		Delisted  bool      `json:"delisted"`
+		Reason    string    `json:"reason"`
+	}
+
+	aliasTrackDelistStatus TrackDelistStatus
+
+	jsonTrackDelistStatus struct {
+		*aliasTrackDelistStatus
+		CreatedAt string `json:"createdAt"`
+	}
+
+	UserDelistStatus struct {
+		CreatedAt time.Time `json:"createdAt"`
+		UserID    int       `json:"userId"`
+		Delisted  bool      `json:"delisted"`
+		Reason    string    `json:"reason"`
+	}
+
+	aliasUserDelistStatus UserDelistStatus
+
+	jsonUserDelistStatus struct {
+		*aliasUserDelistStatus
+		CreatedAt string `json:"createdAt"`
+	}
+)
+
+func (t *TrackDelistStatus) UnmarshalJSON(data []byte) error {
+	temp := &jsonTrackDelistStatus{
+		aliasTrackDelistStatus: (*aliasTrackDelistStatus)(t),
+	}
+	if err := json.Unmarshal(data, &temp); err != nil {
+		return err
+	}
+	var err error
+	t.CreatedAt, err = time.Parse("2006-01-02 15:04:05.999999-07", temp.CreatedAt)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (u *UserDelistStatus) UnmarshalJSON(data []byte) error {
+	temp := &jsonUserDelistStatus{
+		aliasUserDelistStatus: (*aliasUserDelistStatus)(u),
+	}
+	if err := json.Unmarshal(data, &temp); err != nil {
+		return err
+	}
+	var err error
+	u.CreatedAt, err = time.Parse("2006-01-02 15:04:05.999999-07", temp.CreatedAt)
+	if err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
might take a look later to see if there's a cleaner way or if you have an idea off the top of your head @michellebrier . this is because trusted notifier returns a string in a format that go Time doesn't accept by default, so we have to unmarshal it separately to keep microseconds